packages/react-query/src/infiniteQueryOptions.ts

```diff
@@ -20,7 +21,11 @@ export type UndefinedInitialDataInfiniteOptions<
   TQueryKey,
   TPageParam
 > & {
-  initialData?: undefined
+  initialData?:
+    | undefined
+    | InitialDataFunction<
+        NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>
```

Comment:
should this have been something like

```ts
  initialData?:
    | undefined
    | InfiniteData<TQueryFnData, TPageParam>
    | InitialDataFunction<
        NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>
```

to allow passing initialData as object and not within a callback function?

Reply:
we can pass initialData as an object via `DefinedInitialDataInfiniteOptions`:

https://github.com/TanStack/query/blob/febbde676d0888d8dd4868177868a3316686cb58/packages/react-query/src/infiniteQueryOptions.ts#L41-L45

Reply:
sure, but it still won't work conditionally, see this modified playground based on the original example: 

https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgVwM4FMCSA7AZsbYGdARWXSgE8AaOA-Q4siygeTBmAm1TgF84uKBBBwARAAEYAQx4yAxgGsA9FHTT5MALQBHclTEBuAFDHcybJq7YUGAOpRpYMBQAUSAkWDSANgBFpGX4ALkQ6Rm9-QOkAflCkdAAPaXAfdFCYKHJ+fgBKRGM4OHluVHg9FnZOUrgAXnCGIlJ9Ng5rVHdCorgKqgBpdEpQgG0xJJSwNLEAXWouot7KADFsUOlUSks4V3zagD5wr18AmTnuw85j6NDPS6iZee6YsLBpAHN0VBHbyJPpWbgrw+AAVpI4QF84MNpvxHkVQhYACboBjoRFnbofGAAOSSMFBILBKVCOzqBwAjBiij9fAT0KDwaFKV0+LkTHxTOZLNUbGh0A4nC4oMwqO4Lr9oiEwjT7rF4nBxql0nBMtkBKyCkU1DBkFBeRgcI0mC1OudFgMhlCxsklTMqT0Wis1hstqT9uKrqdHjK-jcIp7pHC4M83QcxUDPt9-bKARGGSlIdC8kGEdhkaj0Y8sbjEvj3vSiSASbsKfaZXT40W4Myiqz2UA

Reply:
okay then what’s the fix please 😅

Reply:
I made a PR here :) https://github.com/TanStack/query/pull/8157

Hope that clarifies things, or perhaps I'm misunderstanding something!